plugin/plugin.go

```diff
@@ -0,0 +1,107 @@
+package plugin
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spf13/viper"
+)
+
```

Comment:
All the functions below should be methods of a struct (let's call is Podium).
Then, you can call NewPodium(config *viper.Viper) and it saves config.GetString("podium.url") on a attribute of this struct called URL. You can also save User, Pass and other things.